pkg/ipam/crd.go

```diff
@@ -515,8 +525,40 @@ func (a *crdAllocator) buildAllocationResult(ip net.IP, ipInfo *ipamTypes.Alloca
 
 		result = nil
 		err = fmt.Errorf("unable to find ENI %s", ipInfo.Resource)
+
+	// In AlibabaCloud mode,
+	case ipamOption.IPAMAlibabaCloud:
+		for _, eni := range a.store.ownNode.Status.AlibabaCloud.ENIs {
+			if eni.NetworkInterfaceID != ipInfo.Resource {
+				continue
+			}
+			result.PrimaryMAC = eni.MacAddress
+			result.CIDRs = []string{eni.VSwitch.CidrBlock}
+
+			gw, err2 := metadata.GetENIGateway(context.TODO(), eni.MacAddress)
+			if err2 != nil {
+				err = fmt.Errorf("unable get gw from ENI %w", err2)
+				return
+			}
+			result.GatewayIP = gw.IP.String()
+
+			links, err2 := netlink.LinkList()
+			if err2 != nil {
+				err = fmt.Errorf("unable get link %w", err2)
```

Comment:
```suggestion
				err = fmt.Errorf("unable to get link: %w", err2)
```
Same remark, is `err2` necessary?